### PR TITLE
Become standard-compliant with UGRID and CF 1.7 #131

### LINKED ISSUE #131: Become standard-compliant with UGRID and CF 1.7
follow up of #32 . see comment on https://github.com/lutraconsulting/MDAL/pull/117

- [ ]  Instead of `meshname_that_you_are_loading + "_node_z"` use variable that has `:standard_name="altitude"` and `:location="node"` and `:mesh="meshname_that_you_are_loading"` defined
- [ ] Instead of `meshname_that_you_are_loading :node_dimension` search for the node x variable as can be found in `meshname_that_you_are_loading:node_coordinates` and inquire for its (one and only) dimension 
- [ ] detto for face_dimension 
- [ ] Instead of `max_face_nodes_dimension` use the other dimension of the variable that was specified in `mesh2d:face_node_connectivity` attribute (e.g., variable "mesh2d_face_nodes").

### PR DESCRIPTION
**Pull request by vcloarec**
fix #131

### CODE REVIEW

File: mdal/frmts/mdal_ugrid.cpp [hunk 1]
```diff
@@ -61,33 +61,96 @@ MDAL::CFDimensions MDAL::DriverUgrid::populateDimensions( )
...
-  dims.setDimension( CFDimensions::Time, count, ncid );
+  // not required for UGRID format, so catch the exception to handle with it
+  try
+  {
+    mNcFile.getDimension( "time", &count, &ncid );
+    dims.setDimension( CFDimensions::Time, count, ncid );
+  }
+  catch ( MDAL_Status )
+  {
+    dims.setDimension( CFDimensions::Time, 0, -1 );
```
Comment (PeterPetrik) [inline mdal/frmts/mdal_ugrid.cpp]: Do you have some testcase without time?

File: mdal/frmts/mdal_ugrid.cpp [hunk 2]
```diff
@@ -61,33 +61,96 @@ MDAL::CFDimensions MDAL::DriverUgrid::populateDimensions( )
...
-  // maximum number of vertices in faces
-  const std::string mesh2dMaxNodesInFace = mNcFile.getAttrStr( mMesh2dName, "max_face_nodes_dimension" );
-  mNcFile.getDimension( mesh2dMaxNodesInFace, &count, &ncid );
-  dims.setDimension( CFDimensions::MaxVerticesInFace, count, ncid );
 
   // Time
-  mNcFile.getDimension( "time", &count, &ncid );
-  dims.setDimension( CFDimensions::Time, count, ncid );
+  // not required for UGRID format, so catch the exception to handle with it
+  try
```
Comment (PeterPetrik) [inline mdal/frmts/mdal_ugrid.cpp]: maybe use "hasDimension" instead of try ... catch?

File: mdal/frmts/mdal_ugrid.cpp [hunk 3]
```diff
@@ -61,33 +61,96 @@ MDAL::CFDimensions MDAL::DriverUgrid::populateDimensions( )
...
+  dims.setDimension( CFDimensions::MaxVerticesInFace, maxVerticesPerFace, maxVerticesPerFaceDimensionId );
+
+
+
+  // number of edges in the mesh, not required for UGRID format, so catch the exception to handle with it.
+  //Is it usefull with MDAL ? It seems MDAL don't use edges
   const std::string mesh2dEdge = mNcFile.getAttrStr( mMesh2dName, "edge_dimension" );
-  mNcFile.getDimension( mesh2dEdge, &count, &ncid );
-  dims.setDimension( CFDimensions::Face2DEdge, count, ncid );
+  try
```
Comment (PeterPetrik) [inline mdal/frmts/mdal_ugrid.cpp]: maybe use "hasDimension" instead of try ... catch?

File: mdal/frmts/mdal_ugrid.cpp [hunk 4]
```diff
@@ -61,33 +61,96 @@ MDAL::CFDimensions MDAL::DriverUgrid::populateDimensions( )
...
+    maxVerticesPerFace = faceDimension.at( 1 );
+  }
+
+  dims.setDimension( CFDimensions::Face2D, facesCount, facesIndexDimensionId );
+  dims.setDimension( CFDimensions::MaxVerticesInFace, maxVerticesPerFace, maxVerticesPerFaceDimensionId );
+
+
+
+  // number of edges in the mesh, not required for UGRID format, so catch the exception to handle with it.
+  //Is it usefull with MDAL ? It seems MDAL don't use edges
```
Comment (PeterPetrik) [inline mdal/frmts/mdal_ugrid.cpp]: not yet :)

File: mdal/frmts/mdal_ugrid.cpp (hunk 3 above)
Comment (vcloarec) [reply mdal/frmts/mdal_ugrid.cpp]: I agree, it is cleaner.

File: mdal/frmts/mdal_ugrid.cpp (hunk 2 above)
Comment (vcloarec) [reply mdal/frmts/mdal_ugrid.cpp]: I agree, it is cleaner.

File: mdal/frmts/mdal_ugrid.cpp (hunk 1 above)
Comment (vcloarec) [reply mdal/frmts/mdal_ugrid.cpp]: I don't know any existing UGRID file format without time ... But very soon, I will have one : I am working on saving TIN mesh with UGRID format in my project (https://github.com/vcloarec/ReosProject/issues/1#issuecomment-517055478).
As TIN doesn't have time variable, those files will have to pass a test.

**Pull request merged**